core/src/main/java/org/geysermc/geyser/entity/type/player/PlayerEntity.java

```diff
@@ -431,11 +465,14 @@ public void setBelowNameText(Objective objective) {
                 packet.getMetadata().put(EntityDataTypes.SCORE, displayString);
                 session.sendUpstreamPacket(packet);
             }
-        } else if (valid) {
-            SetEntityDataPacket packet = new SetEntityDataPacket();
-            packet.setRuntimeEntityId(geyserId);
-            packet.getMetadata().put(EntityDataTypes.SCORE, "");
-            session.sendUpstreamPacket(packet);
+        } else {
+            dirtyMetadata.put(EntityDataTypes.SCORE, "");
```

Comment:
When `PlayerEntity#setBelowNameText` is called from `PlayerEntity#spawnEntity`, `valid` is still `false`, so the `SetEntityDataPacket` is not sent. This meant that the below name text would not appear from joining the world until it was next updated. Changes to `dirtyMetadata` get picked up by `PlayerEntity#spawnEntity` and then get included in the `AddPlayerPacket`.

This was also included in the old version of `JavaSetScoreTranslator#setBelowName` (which I changed to call `PlayerEntity#setBelowNameText` to avoid having the same code in two places)
https://github.com/GeyserMC/Geyser/blob/a24f68412315fdabe93053d7722f9c9f15fc6511/core/src/main/java/org/geysermc/geyser/translator/protocol/java/scoreboard/JavaSetScoreTranslator.java#L90-L104

If there's a better way of doing this, please let me know

Reply:
We can probably move this to an `else` statement then after the `if (valid)` check below.

Reply:
Done :)